src/Microsoft.Health.Fhir.Shared.Api/Modules/SearchModule.cs

```diff
@@ -77,7 +86,31 @@ public void Load(IServiceCollection services)
                 .AsSelf()
                 .AsService<IFhirElementToSearchValueTypeConverterManager>();
 
-            services.AddSingleton<ISearchIndexer, SearchIndexer>();
+            // TypedElement based converters
+            services.TypesInSameAssemblyAs<IFhirNodeToSearchValueTypeConverter>()
+                .AssignableTo<IFhirNodeToSearchValueTypeConverter>()
+                .Singleton()
+                .AsService<IFhirNodeToSearchValueTypeConverter>();
+
+            services.Add<FhirNodeToSearchValueTypeConverterManager>()
+                .Singleton()
+                .AsSelf()
+                .AsService<IFhirNodeToSearchValueTypeConverterManager>();
+
+            services.Add<CodeSystemResolver>()
+                .Singleton()
+                .AsSelf()
+                .AsImplementedInterfaces();
+
+            if (_configuration.CoreFeatures.UseTypedElementIndexer)
```

Comment:
Would it make sense to add all the services used by the typed element indexer to this if statement, and similarly the services used by the old search indexer to the else statement? I think it would improve readability and not add some services that wouldn't be used.

Reply:
IFhirNodeToSearchValueTypeConverter need to always be registered as they are also used by the SearchParameterSupportResolver. I can move the IFhirElementToSearchValueTypeConverter based converters